app/Tag.php

```diff
@@ -1,15 +0,0 @@
-<?php
-
-namespace App;
-
-use Illuminate\Database\Eloquent\Model;
-
-class Tag extends Model
-{
-    protected $guarded = [];
-
-    public function posts()
```

Comment:
You could also look into Polymorphic relations. Not necessary in this setup, but could be nice to have if you plan on having more things you could tag.